bc-backend/src/main/java/org/jboss/da/bc/impl/ProjectHiearchyCreator.java

```diff
@@ -209,20 +226,30 @@ private void setSCMInfo(ProjectDetail project, Optional<POMInfo> pomInfo) {
 
         if (!url.isPresent() || !rev.isPresent()) {
             try {
-                boolean gavInRepository = scm.isGAVInRepository(scmUrl, revison, project.getGav());
+                String scmUrlTemp;
+                String revisionTemp;
+                if (revision == null) {
+                    scmUrlTemp = externalScmUrl;
+                    revisionTemp = externalRevison;                    
+                }
+                else {
+                    scmUrlTemp = scmUrl;
+                    revisionTemp = revision;
+                }
+                boolean gavInRepository = scm.isGAVInRepository(scmUrlTemp, revisionTemp, project.getGav());
                 if (gavInRepository) {
                     if (url.isPresent()) {
-                        if (url.get().equals(scmUrl)) {
-                            rev = Optional.of(revison);
+                        if (url.get().equals(scmUrlTemp)) {
+                            rev = Optional.of(revisionTemp);
                         }
                     } else {
                         if (rev.isPresent()) {
-                            if (rev.get().equals(revison)) {
-                                url = Optional.of(scmUrl);
+                            if (rev.get().equals(revisionTemp)) {
+                                url = Optional.of(scmUrlTemp);
                             }
                         } else {
-                            url = Optional.of(scmUrl);
-                            rev = Optional.of(revison);
+                            url = Optional.of(scmUrlTemp);
+                            rev = Optional.of(revisionTemp);
                         }
                     }
                 }
```

Comment:
You should try and consider where the scm url was taken from (external internal) and at the end set it properly.
 * If the gav is in repository
   * we should copy the coordinates (internal and external) from the toplevel
 * if the gav is not in repository but there is coordinates in the pom
   * we should consider the url from pom to be external url
 * if the gav is not in repository and there are no coordinates in the pom
  * both internal and external url are null